GCMtools/utils/interface.py

```diff
@@ -10,27 +10,344 @@
+        """
+        from petitRADTRANS.poor_mans_nonequ_chem import interpol_abundances
+
+        if c['time'] in self.ds.dims:
+            raise ValueError('Dataset should not have a timedimension. Select the timestamp beforehand.')
+
+        p = self.ds[c['Z']].values
+        p_unit = self.ds.attrs.get('p_unit')
+        if p_unit == 'Pa':
+            p = p / 1e5  # convert to bar
+        if p_unit not in ['Pa', 'bar']:
+            raise NotImplementedError('can currently only deal with Pa or bar')
+
+        COs = np.ones_like(p) * CO
+        FeHs = np.ones_like(p) * FeH
+
+        var_names = interpol_abundances(np.array([0.55]), np.array([0.0]), np.array([100]), np.array([0.1])).keys()
+        for key in [c['T'], *var_names]:
+            self.abunds.update({key: xr.DataArray(
+                data=np.empty((len(self.ds[c['lon']]), len(self.ds[c['lat']]), len(self.ds[c['Z']]))),
+                dims=[c['lon'], c['lat'], c['Z']],
+                coords={c['lon']: self.ds[c['lon']],
+                        c['lat']: self.ds[c['lat']],
+                        c['Z']: self.ds[c['Z']]})})
+
+        for lon in self.ds.lon:
+            for lat in self.ds.lat:
+                temp_i = self.ds[temp_key].sel(lon=lon, lat=lat)
+                abus = interpol_abundances(COs, FeHs, temp_i, p)
+                for sp in var_names:
+                    self.abunds[sp].loc[{c['lon']: lon, c['lat']: lat}] = abus[sp]
+                self.abunds[c['T']].loc[{c['lon']: lon, c['lat']: lat}] = temp_i
+
+        self.abunds.attrs.update({'p_unit': p_unit})
+
+    def to_pRT(self, prt_species, p_prt):
+        """
+        Format chemistry to match petitRADTRANS requirements.
+
+        Parameters
+        ----------
+        prt_species: list
+            List of species names used in petitRADTRANS
+        p_prt: list
+            Pressure in bar, as used in petitRADTRANS
+
+        Returns
+        -------
+        prt_abu: Dataset
+            abundances ready for prt
+        """
+        prt_abu = self.abunds.copy()
+        for sp_raw in prt_species:
+            sp = sp_raw.split('_')[0]
+            if sp not in self.abunds:
+                raise ValueError(f'We miss chemistry data for {sp}')
+
+            prt_abu.update({sp_raw: self.abunds[sp]})
+
+        p_unit = self.abunds.attrs.get('p_unit')
+        if p_unit not in ['Pa', 'bar']:
+            raise NotImplementedError('can currently only deal with Pa or bar')
+
+        if p_unit == 'Pa':
+            p_prt = p_prt * 1e5
+
+        prt_abu = prt_abu.interp(Z=p_prt)
+
+        return prt_abu
+
+
 class Interface:
     """
     The GCMtools interfacing class which implements common
     interfacing functionalities to different codes.
 
-    Attributes
-    ----------
-
     Methods
     -------
+    set_data: Function that handles the input from GCMT
     """
 
+    chemistry = Chemistry()
```

Comment:
This needs to go into the constructor, otherwise it is a 'class variable' instead of an 'instance variable'.